cellengine/resources/fcs_file.py

```diff
@@ -219,6 +219,14 @@ def update(self):
     def delete(self):
         return ce.APIClient().delete_entity(self.experiment_id, "fcsfiles", self._id)
 
+    def save(self, filepath: str):
```

Comment:
This isn't quite the API we need. `get_events` covers saving the data from CellEngine as an FCS file. This function needs to cover creating an FCS file from scratch or based on an existing one, then saving it to disk or uploading to CellEngine. File data in CellEngine is immutable, so it will always be uploading a new file. I'm realizing that we have no interface for modifying header keyword values in the toolkit, which is pretty important; `header` is just a JSON string currently.

The `create_fcs` call also needs to pass reagents to `opt_channel_names`.

If you want to land this PR quickly, you could remove this fn and we can work on this part separately.

Reply:
What exactly do you mean by "from scratch"; do we want to be able to create a new fcs file from an arbitrary (correctly-formatted) dataframe, or..?

Reply:
Yes, from a dataframe + list of channel names and optionally reagents.

Reply:
Most recent changes cover channel reagents and uploading a df, but not yet updating header values. For that, would you like the header represented by another Python object with getters and setters?

Reply:
Because they're big, you have to request it with `fields=+header`.